### PR TITLE
Facilitate C-binding for Go

### PR DESCRIPTION
**Pull request by renukamanavalan**
Added a C-Wrap to facilitate GO binding.
Simplified event_receive  signature to make python binding easier.

### CODE REVIEW

File: common/events.cpp [hunk 1]
```diff
@@ -418,24 +419,170 @@ events_deinit_subscriber(event_handle_t handle)
...
+        for (auto it = data.cbegin(); it != data.cend(); ++it) {
+            if ((it.key() == ARGS_USE_CACHE) && (*it).is_boolean()) {
+                use_cache = it.value();
+            }
+            else if ((it.key() == ARGS_RECV_TIMEOUT) && (*it).is_number_integer()) {
+                recv_timeout = it.value();
+            }
+        }
+    }
+    void *h = events_init_subscriber(use_cache, recv_timeout);
```
Comment (zbud-msft) [inline common/events.cpp]: Avoid one letter var names

File: common/events.cpp [hunk 2]
```diff
@@ -418,24 +419,170 @@ events_deinit_subscriber(event_handle_t handle)
...
+
+        int rc_missed = snprintf(missed_cnt_str, missed_cnt_str_sz, "%d", evt.missed_cnt);
+        if (rc_missed >= missed_cnt_str_sz) {
+            SWSS_LOG_ERROR("missed cnt (%d) buffer.need=%d given=%d",
+                    evt.missed_cnt, rc_missed, missed_cnt_str_sz);
+            missed_cnt_str[missed_cnt_str_sz-1] = 0;
+        }
+    }
+    else if (evt.rc > 0) {
+        // timoeut
```
Comment (zbud-msft) [inline common/events.cpp]: Spelling mistake

File: common/events.cpp [hunk 3]
```diff
@@ -418,24 +419,170 @@ events_deinit_subscriber(event_handle_t handle)
...
+        }
+
+        int rc_missed = snprintf(missed_cnt_str, missed_cnt_str_sz, "%d", evt.missed_cnt);
+        if (rc_missed >= missed_cnt_str_sz) {
+            SWSS_LOG_ERROR("missed cnt (%d) buffer.need=%d given=%d",
+                    evt.missed_cnt, rc_missed, missed_cnt_str_sz);
+            missed_cnt_str[missed_cnt_str_sz-1] = 0;
+        }
+    }
+    else if (evt.rc > 0) {
```
Comment (zbud-msft) [inline common/events.cpp:585]: What are the cases when evt.rc > 0?

File: common/events.cpp (hunk 2 above)
Comment (renukamanavalan) [reply common/events.cpp]: fixed

File: common/events.cpp (hunk 3 above)
Comment (renukamanavalan) [reply common/events.cpp:585]: timeout only.

**Review by zbud-msft — APPROVED** [review]
lgtm

File: common/events.cpp (hunk 1 above)
Comment (renukamanavalan) [reply common/events.cpp]: Fixed.

**Pull request merged**